Python/bltinmodule.c

```diff
@@ -1465,21 +1523,39 @@ map_reduce(mapobject *lz, PyObject *Py_UNUSED(ignored))
         PyTuple_SET_ITEM(args, i+1, Py_NewRef(it));
     }
 
+    if (lz->strict) {
+        return Py_BuildValue("ONO", Py_TYPE(lz), args, Py_True);
+    }
     return Py_BuildValue("ON", Py_TYPE(lz), args);
 }
 
+PyDoc_STRVAR(setstate_doc, "Set state information for unpickling.");
+
+static PyObject *
+map_setstate(mapobject *lz, PyObject *state)
+{
+    // For now we keep things strict, rather than using PyObject_IsTrue().
```

Comment:
This comment might seem contradictory with the constructor which computes the object truthiness. So I think you should do the same as it's done in `zip` or make the constructor reject non-boolean values.

Reply:
Hmm, let's revert that change then. We would want `zip()` and `map()` to act the same anyway when strict is set.